lovefield/lovefield.d.ts

```diff
@@ -158,12 +158,14 @@ declare namespace lf {
       getNormalizedName(): string
     }
 
-    export interface Table {
+    interface ITable {
       as(name: string): Table
       createRow(value: Object): Row
       getName(): string
     }
 
+    export type Table = ITable & { [index: string]: Column }
```

Comment:
Don't fully understand this type here. Can you explain what does `{ [index: string]: Column }` mean? An example compiler error and subsequent success with this change would be helpful.